pkg/utils/yaml_utils.go

```diff
@@ -221,3 +227,29 @@ func IsYAML(data string) bool {
 
 	return isMap || isSlice
 }
+
+// GetGitRoot returns the root directory of the Git repository using go-git.
+func GetGitRoot() (string, error) {
```

Comment:
This function doesn't seem related to YAML. Let's move it to utils/git.go and aim for smaller files. We are not good about "small files", but that doesn't mean we cannot start =)